File: integration-tests/hibernate-reactive-panache/src/main/java/io/quarkus/it/panache/reactive/NoPagingTestEndpoint.java
package io.quarkus.it.panache.reactive;

import jakarta.ws.rs.GET;
import jakarta.ws.rs.Path;

import io.quarkus.hibernate.reactive.panache.Panache;
import io.smallrye.mutiny.Uni;

/**
 * Run a simple, no paged PanacheQueryTest in order to log the generated SQL.
 *
 * @see io.quarkus.it.panache.reactive.NoPagingPMT
 */
@Path("no-paging-test")
public class NoPagingTestEndpoint {

    @GET
    public Uni<String> test() {
        return Panache.withTransaction(() -> PageItem.findAll().list().map(v -> "OK"));
    }
}

File: integration-tests/hibernate-reactive-panache-kotlin/src/main/kotlin/io/quarkus/it/panache/reactive/kotlin/NoPagingTestEndpoint.kt
package io.quarkus.it.panache.reactive.kotlin

import io.quarkus.hibernate.reactive.panache.Panache
import io.smallrye.mutiny.Uni
import jakarta.ws.rs.GET
import jakarta.ws.rs.Path

/**
 * Run a simple, no paged PanacheQueryTest in order to log the generated SQL.
 *
 * @see io.quarkus.it.panache.reactive.kotlin.NoPagingPMT
 */
@Path("no-paging-test")
class NoPagingTestEndpoint {
    @GET
    fun test(): Uni<String> {
        return Panache.withTransaction { PageItem.findAll().list().map { v -> "OK" } }
    }
}
